pyam/core.py

```diff
@@ -510,14 +517,125 @@ def check_aggregate(self, variable, components=None, units=None,
+                'regional components'
+            )
+            logger().info(msg.format(variable))
+
+            return None
+
+        # filter and groupby data, use `pd.Series.align` for matching index
+        df_region, df_components = (
+            _aggregate_by_regions(var_df.data, region, units)
+            .align(_aggregate_by_regions(var_df.data, components, units))
+        )
+
+        # if it's a World comparison, we need to add in variables that might
+        # have been missed e.g. shipping and aviation
+        if region == "World":
+            df_components.index = df_components.index.droplevel(
+                "variable"
+            )
+            non_world_region = components[0]
+
+            for var_to_add in self.filter(variable="{}|*".format(variable)).variables():
+
+                var_has_regional_info = (
+                    (self.data.variable == var_to_add)
+                    & (self.data.region == non_world_region)
+                ).any()
+
+                if not var_has_regional_info:
+                    df_var_to_add = self.filter(region=region, variable=var_to_add).data.groupby(REGION_IDX).sum()['value']
+                    df_var_to_add.index = df_var_to_add.index.droplevel("variable")
+
+                    if len(df_var_to_add):
+                        df_components = df_components.add(df_var_to_add,
+                                                          fill_value=0)
+
+            df_components = pd.concat([df_components], keys=[variable],
+                                      names=['variable'])
+
+        # use `np.isclose` for checking match
+        diff = df_region[~np.isclose(df_region, multiplier * df_components,
+                                     **kwargs)]
+
+        if len(diff):
+            msg = (
+                '{} - {} of {} data points are not aggregates of regional '
+                'components'
+            )
+            logger().info(msg.format(variable, len(diff), len(df_region)))
+
+            if exclude_on_fail:
+                self._exclude_on_fail(diff.index.droplevel([2, 3]))
+
+            diff = pd.concat([diff], keys=[region], names=['region'])
+
+            return diff.unstack().rename_axis(None, axis=1)
+
+    def check_internal_consistency(self, **kwargs):
+        #TODO: test this and write docstring, kwargs passed to np is close
+        inconsistent_vars = {}
+        for variable in self.variables():
+            diff_agg = self.check_aggregate(
+                variable,
+                **kwargs
+            )
+            if diff_agg is not None:
+                inconsistent_vars[variable + "-aggregate"] = diff_agg
+
+            diff_regional = self.check_aggregate_regions(
+                variable,
+                **kwargs
+            )
+            if diff_regional is not None:
+                inconsistent_vars[variable + "-regional"] = diff_regional
+
+        if inconsistent_vars:
+            return inconsistent_vars
```

Comment:
Since you define it initially as an empty dict, I was just return it (and not check)